doc/lib/mqtt.txt

```diff
@@ -0,0 +1,60 @@
+/**
+@mainpage
+@anchor mqtt
+@brief MQTT 3.1.1 client library.
+
+MQTT is a standardized publish subscribe protocol popular on embedded devices. The MQTT library provided with this SDK implements a subset of the MQTT v3.1.1 protocol. The sample application for Linux `subscribe_publish_sample.c` is provided as a starting point for using this library.
+
+The interface for communication over MQTT is provided in the file `aws_iot_mqtt_interface.h`.
+- MQTT client context management: @ref mqtt_function_init and @ref mqtt_function_free
+- Connection management: @ref mqtt_function_connect and @ref mqtt_function_disconnect
+- Publishing messages to the server: @ref mqtt_function_publish
+- Managing subscriptions: @ref mqtt_function_subscribe and @ref mqtt_function_unsubscribe
+- Process incoming messages, reconnections, and keep-alive: @ref mqtt_function_yield
+
+@note In a multithreaded environment, always acquire a lock before calling any function in this library. This library is not thread safe.
+
+@note This library does not support QoS 2 or retained messages.
+
+@section mqtt_configuration Configuration
+@brief The following configuration settings are associated with this MQTT library.
+- `AWS_IOT_MQTT_TX_BUF_LEN` <br>
+Size of buffer for outgoing messages.
+- `AWS_IOT_MQTT_RX_BUF_LEN` <br>
+Size of buffer for incoming messages. Messages longer than this will be dropped.
+- `AWS_IOT_MQTT_NUM_SUBSCRIBE_HANDLERS` <br>
+Number of subscriptions that may be registered simultaneously.
+- `AWS_IOT_MQTT_MIN_RECONNECT_WAIT_INTERVAL` <br>
+The initial wait time before the first reconnect attempt. See @ref mqtt_autoreconnect.
+- `AWS_IOT_MQTT_MAX_RECONNECT_WAIT_INTERVAL` <br>
+The maximum wait time between reconnect attempts. See @ref mqtt_autoreconnect.
+*/
+
+/**
+@page mqtt_autoreconnect Auto-reconnect
+@brief This page described the auto-reconnect mechanism of the MQTT library.
+
+The auto-reconnect mechanism can be enabled through the following methods:
+- Set #IoT_Client_Init_Params.enableAutoReconnect to `true` when calling @ref mqtt_function_init.
+- Call @ref mqtt_function_autoreconnect_set_status with `newStatus` set to `true`. @ref mqtt_function_autoreconnect_set_status must only be called after @ref mqtt_function_connect.
+
+Once enabled, this feature will attempt to reconnect to the server if the library detects a disconnection. The reconnections are handled through @ref mqtt_function_yield. In addition to re-establishing the network connection and MQTT session, topic filter subscriptions registered with the client will be resubscribed.
+
```

Comment:
Is the re-subscription on re-conenction only for non-persistent sessions? If yes, would be good to call that out here

Reply:
This library only supports non-persistent sessions.  Therefore, I don't think we need to specify.

Reply:
Is that explicitly mentioned anywhere in the doc?

Reply:
Never mind, I took a closer look and this version allows persistent sessions.  Will update wording.